main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Projects.FileNesting/FileNestingService.cs

```diff
@@ -207,6 +203,24 @@ ProjectFileNestingInfo AddFile (ProjectFile projectFile)
 				}
 			}
 
+			// Check existing files for children of the newly added file
+			foreach (var file in Project.Files.GetFilesInFolder (projectFile.FilePath.ParentDirectory)) {
+				if (file == projectFile)
+					continue;
+
+				var isParent = FileNestingService.InternalGetParentFile (file) == projectFile;
+				if (isParent) {
+					var child = projectFiles.GetOrAdd (file, new ProjectFileNestingInfo (file));
```

Comment:
```suggestion
					var child = projectFiles.GetOrAdd (file, file => new ProjectFileNestingInfo (file));
```

Otherwise we'll always allocate that `ProjectFileNestingInfo`

Reply:
And given there's no `break`, I assume this can be entered multiple times.